apps/cli/src/commands/build.ts

```diff
@@ -0,0 +1,296 @@
+import { Command, Flags } from "@oclif/core";
+import fs from "fs-extra";
+import path from "path";
+import { execa } from "execa";
+import tmp from "tmp";
+import bytes from "bytes";
+
+type ImageBuildOptions = {
+    network: string;
+    target?: string;
+};
+
+type ImageInfo = {
+    cmd: string[];
+    dataSize: string;
+    entrypoint: string[];
+    env: string[];
+    ramSize: string;
+    sdkVersion: string;
+    workdir: string;
+};
+
+const CARTESI_LABEL_PREFIX = "io.cartesi.rollups";
```

Comment:
I'm not sure if we need a machine namespace/hierarchy like `io.cartesi.rollups.machine`.

Imagine if we need to control other components from the cartesi-node that are related to the DApp image and need to be passed to, say, `dispatcher`, like the `network` (chain) or something else.

`io.cartesi.rollups.node.network` could hold this information.

I know we can change this in the future, won't block for it, but I think it deserved some thought/design.

Reply:
I understand.

The network is a trickier example, that we will need to tackle soon. Because it will need to produce different machines.